src/planner.cc

```diff
@@ -38,17 +47,77 @@ namespace {
 // The speed is initially the aimed goal; if it cannot be reached, the
 // AxisTarget will be modified to contain the actually reachable value. That is
 // used in planning along the path.
+// The starting_speed is the required speed at the start of the segment.
+// It is used to combine segments with different trajectory angles.
+// In the AxisTarget we define "motion invariants" (or targets). Values and
+// parameters that assuming the requested final trajectory stays the same, they
+// are constants, no matter the previous or next segments.
 struct AxisTarget {
   int position_steps[GCODE_NUM_AXES];  // Absolute position at end of segment.
```

Comment:
Suggest to use
```
AxesRegister position_steps;
AxesRegister delta_steps;
```
instead of he raw arrays.
(would require to change the memset with call to AxesRegister::zero() and other places that use a pointer to use a `const AxesRegister&` instead.

Reply:
(mmh, AxesRegister is float, so we need `FixedArray<int, GCODE_NUM_AXES, GCodeParserAxis>`.)

Reply:
(solved)